src/Surface.zig

```diff
@@ -1833,6 +1833,22 @@ pub fn keyCallback(
     return .consumed;
 }
 
+pub fn isBindingMaskedByKittyDisambiguate(
+    self: *Surface,
+    action: input.Binding.Action,
+) bool {
+    switch (action) {
+        .esc_unless_kitty_disambiguate,
+        .text_unless_kitty_disambiguate => {
+            if (self.io.terminal.screen.kitty_keyboard.current().disambiguate) {
```

Comment:
For future note: this requires holding "the big lock" (expensive and slows everything down). I don't think the lock is held here and introducing it in this path kind of sucks. Its not the worst since its gated on the binding action but I still don't love it.